Improvements (#60)
nodeshift/kube-service-bindings: README.md

### type

The type of service for which a binding is being
requested. Currently the supported types are:

- 'KAFKA'
- 'POSTGRESQL'
- 'REDIS'
- 'MONGODB'
- 'AMQP'
- 'MYSQL'

### client

The package the application is using to connect
to the service. kube-service-bindings is aware of a
subset of possible packages. For those that it is aware
of, it can map the service bindings into the form
required by the client.

Currently the following clients are recognized based on the supported types:

- KAFKA
  - node-rdkafka
  - kafkajs
- POSTGRESQL
  - pg
  - odbc
- REDIS
  - redis
  - ioredis
- MONGODB
  - mongodb
  - mongoose
- AMQP
  - rhea
- MYSQL
  - mysql
  - mysql2
  - odbc

**(Deprecated)** If you don't specify a client, the object returned will
be a direct map from the bindings, with the keys
corresponding to the name of each file provided by the
binding.

#### Example on mongoDB client

```javascript
const serviceBindings = require('kube-service-bindings');
const { MongoClient } = require('mongodb');

const { url, connectionOptions } = serviceBindings.getBinding(
  'MONGODB',
  'mongodb'
);

const mongoClient = new MongoClient(url, connectionOptions);
```

### bindingOptions

> An object where provides control on parsing binding data.

Reviewer: ```suggestion
An object which provides additional control over how binding data is parsed.
```

| Attribute                         | type      | default Value |
| --------------------------------- | --------- | ------------- |
| [id](#id)                         | `String`  | undefined     |

> | [removeUnmapped](#removeunmapped) | `Boolean` | true          |

Reviewer: ```suggestion
| [removeUnmapped](#removeunmapped) | `Boolean` | as set by client, or true if not set by client |
```

> | [allwCopy](#allowcopy)            | `Boolean` | false         |

Reviewer: ```suggestion
| [allowCopy](#allowcopy)            | `Boolean` | false         |
```

| [bindingData](#bindingdata)       | `Object`  | undefined     |

#### id

> Option id that is used to filter the available bindings in the search. Most useful for testing where you might have more than one binding of a given type available.

Reviewer: ```suggestion
Id used to filter the available bindings.  For example, if you have two Kafka services bound to your application an id can be specified to identify which one should be used.  If there are multiple bindings that satisfy a request and no id is specified, the first one found will be used.
```

#### removeUnmapped

> Fine grains parsed binding data to fit for the specified client, removing unnecessary properties from the input object.

Reviewer: ```suggestion
Removes all binding data which is not mapped for the client. Otherwise any binding data which is not mapped remains in it's raw for on the binding object returned.  The default depends on the client.
```

#### allowCopy

> Enables setting proper permissions for some of the binding data, where the system has not provided them correctly. It allows binding files content to be copied/stored in a new file and directory. This has to be enabled by the user in order to be aware of the security risk, as some files might include sensitive material. E.x. connecting to postgresql with odbc client, throws the following error for the tls.key file `permissions should be u=rw (0600) or less`.

Reviewer: ```suggestion
Enables setting proper permissions for some of the binding data, where the system has not provided them correctly. It allows binding files content to be copied/stored in a new file and directory. This has to be enabled by the user in order to be aware of the security risk, as some files might include sensitive material. For example, connecting to postgresql with the odbc client, the following error is thrown  for the tls.key file if copies are not allowed: `permissions should be u=rw (0600) or less`.
```

#### bindingData

> An object for passing binding data to kube-service-bindings. This is useful especially in local dev environtment or as a fallback in case of binding data are not available.

Reviewer: ```suggestion
An optional object for passing binding data to kube-service-bindings. This is useful especially in local dev environtment or as a fallback in case of binding data are not available.
```

```javascript
const serviceBindings = require('kube-service-bindings');

let kafkaConnectionBindings;

try {
  kafkaConnectionBindings = serviceBindings.getBinding('KAFKA', 'kafkajs');
} catch (err) {
  kafkaConnectionBindings = serviceBindings.getBinding('KAFKA', 'kafkajs', {
    bootstrapServers: 'test-boostrap:443',
    clientId: 'client1',
    clientSecret: 'pass1',
    password: 'pass1',
    provider: 'rhoas',
    saslMechanism: 'PLAIN',
    securityProtocol: 'SASL_SSL',
    type: 'kafka',
    user: 'user1'
  });
}
```

  <!--
  | case | Arg1 | Arg2        | Arg3        | Supported | deprecate |
  | ---- | ---- | ----------- | ----------- | --------- | --------- |
  | 1    | -    | -           | -           | true      | -         |
  | 2    | type | -           | -           | true      | true      |
  | 3    | type | id          | -           | false     |           |
  | 4    | type | bindOptions | -           | false     |           |
  | 5    | type | client      | -           | true      | -         |
  | 6    | type | client      | id          | true      | true      |
  | 7    | type | client      | bindOptions | true      | -         | -->

## getBinding()

> By not passing any argument to getBinding function, will return binding data in raw format.

Reviewer: ```suggestion
If you don't specify any parameters the getBinding function will return binding data in raw format.
```
